### DISCUSSION TITLE
Roadmap: Cloud Hosting Platform

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#583] [Roadmap]
### Description

Build a one-click deployment platform specifically for Payload which should be the easiest, most cost-effective, and efficient way to deploy the CMS for production. 

Payload will still be able to be deployed wherever you want on your own completely free of charge.

### Implementation Detail

- AWS to start, more cloud platforms supported in the future
- Automatic database provisioning, including automated backups and third-party access
- Permanent file storage
- Branch-based auto deploys
- Configurable environment variables
- Auto-scaling infrastructure

### Potential Breaking Changes

None

### Effort

High

**Comment by richardvanbergen** [comment]
Hey guys I just saw this pop-up in my feed and I feel like as a developer it'd be much easier for my to advocate for you guys If I could see "Version 1.0.0". This is absolutely the direction you need to take (and I would also advocate for open sourcing at the same time) but I would also like to be able to say Payload is stable. Do we have a roadmap for this?

**Reply by jmikrut** [reply]
Hey @richardvanbergen — hope all is well!

You're 100% right. **_Long_** before we release our cloud platform, we'll be releasing `1.0.0`, including more E2E Cypress tests and a few more features that we've got in the pipeline.

Getting to `1.0.0` is our next target. After that, we'll be working on the cloud platform. 

Good call! 😎

**Comment by etmartinkazoo** [comment]
@jmikrut - We use Hatchbox (https://www.hatchbox.io/) for some Rails apps we maintain and it is phenomenal. It might be a model from which you could draw inspiration. We happily pay the $100 a month for the orchestration and management it provides.

**Comment by ghost** [comment]
Hey @jmikrut, something I really like is the way I can keep Payload CMS in my own cloud infrastructure. It could be cool to create a dedicated AWS VPC to the one-click deployment letting us create peering connections between our VPC in your infrastructure to our VPC in our infrastructure (just like MongoDB Atlas).

**Comment by miguderp** [comment]
It would be fantastic if it could also create temporary staging environments on PRs’ creations (+ duplication of data or seeding provided through configs) and delete them when the PR is merged to the main branch.

I believe Railway does something similar ([Ephemeral Environments](https://docs.railway.app/develop/environments#ephemeral-environments)), I have yet to test it myself

**Comment by johannesschaffer** [comment]
Will it be possible to bring an existing (MongoDB Atlas) database when changing from self hosting to your official hosting?

**Comment by matthijs166** [comment]
I think that Digital Ocean is really easy to add. (https://docs.digitalocean.com/products/app-platform/how-to/add-deploy-do-button/)

Only thing you need is Digital Ocean spaces (S3 storage) & a managed database cluster.

We got one payload instance running at the moment in the DO app platform.

**Comment by OlexBeatable** [comment]
Did someone tried to deploy the Payload CMS to Google Cloud App Engine? I used custom environment with Dockerfile. And in the end if deployment it fails on health check. Could someone help?
